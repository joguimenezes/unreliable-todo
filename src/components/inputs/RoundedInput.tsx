import React from 'react';
import styled from 'styled-components';
import COLORS from '../../utils/constants/color.constant';
import MEDIA_QUERIES from '../../utils/constants/mediaQuery.constant';

type RoundedInputProps = {
  className?: string,
  label?: string,
  labelColor?: string,
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => string | void,
  placeholder?: string,
  testId?: string,
  value?: string | number,
};

const RoundedInput = ({
  className,
  label,
  labelColor,
  onChange,
  placeholder,
  testId,
  value,
}: RoundedInputProps) => (
    <Wrapper>
      <Label labelColor={labelColor}>{label}</Label>
      <Input
        className={className}
        data-testid={testId}
        onChange={onChange}
        placeholder={placeholder}
        value={value}
      />
    </Wrapper>
);

type LabelProps = {
  labelColor?: string,
};

const Label = styled.label`
  color: ${({ labelColor }: LabelProps) => labelColor || COLORS.WHITE};
  font-weight: bold;
  margin-bottom: 5px;
`;

const Wrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  flex-direction: column;
  width: 100%;
`;

const Input = styled.input`
  background-color: transparent;
  border-radius: 6px;
  border: 1px solid ${COLORS.WHITE};
  color: ${COLORS.WHITE};
  font-size: 14px;
  height: 55px;
  margin-bottom: 20px;
  padding-left: 15px;
  width: calc(100% - 45px);

  @media ${MEDIA_QUERIES.DESKTOP_SCREEN} {
    width: 370px;
  }
`;

export default RoundedInput;