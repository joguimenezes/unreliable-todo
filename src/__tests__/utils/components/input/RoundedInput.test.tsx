import React from 'react';
import { render } from '@testing-library/react';
import RoundedInput from '../../../../components/input/RoundedInput';

describe('RoundedInput tests', () => {
  describe('should render correctly', () => {
    it('when has onChange and value', () => {
      const props = {
        value: 100,
        onChange: jest.fn(),
      };

      const { container } = render(<RoundedInput {...props}/>);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('when has placeholder, onChange and value', () => {
      const props = {
        onChange: jest.fn(),
        placeholder: 'Placeholder',
        value: 'Name',
      };

      const { container } = render(<RoundedInput {...props}/>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});