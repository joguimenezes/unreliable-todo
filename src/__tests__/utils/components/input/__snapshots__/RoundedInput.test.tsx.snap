// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundedInput tests should render correctly when has onChange and value 1`] = `
<div
  class="sc-bwzfXH cyfaGt"
>
  <label
    class="sc-bdVaJa iIXUxH"
  />
  <input
    class="sc-htpNat gtqAat"
    value="100"
  />
</div>
`;

exports[`RoundedInput tests should render correctly when has placeholder, onChange and value 1`] = `
<div
  class="sc-bwzfXH cyfaGt"
>
  <label
    class="sc-bdVaJa iIXUxH"
  />
  <input
    class="sc-htpNat gtqAat"
    placeholder="Placeholder"
    value="Name"
  />
</div>
`;
